buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -408,7 +407,17 @@ public int numDirectArenas() {
      * Return the number of thread local caches used by this {@link PooledByteBufAllocator}.
      */
     public int numThreadLocalCaches() {
-        return threadCache.caches.get();
+        PoolArena<?>[] arenas = heapArenas != null ? heapArenas : directArenas;
```

Comment:
@normanmaurer if you look at the code right now [1], it counts two thread caches for heap and direct memory as a single thread cache. That's what I am doing here as well. If both heap and direct arenas exist, both have the same number of tcaches, so I only count one. If one of them doesn't exist, I count the other. If none exist, I return 0.

https://github.com/netty/netty/blob/4.1/buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java#L351